client/src/main/java/io/pravega/client/stream/impl/ReaderGroupStateManager.java

```diff
@@ -462,7 +470,25 @@ void checkpoint(String checkpointName, PositionInternal lastPosition) throws Rea
     boolean isCheckpointSilent(String atCheckpoint) {
         return sync.getState().getCheckpointState().isCheckpointSilent(atCheckpoint);
     }
-    
+
+    @SneakyThrows(CheckpointFailedException.class)
```

Comment:
This is not a good use of sneakythrows. (See our contributing guidelines)

Reply:
This exception is thrown as a SneakyThrows in all places where we attempt to get the StreamCut from a checkpoint. 
As per Contributing guidelines SnealyThrows should be used "_For the specific case where you know with certainty an exception cannot occur_" and this is that case.

Reply:
Oh. I didn't realize it was impossible. In that case, it's fine.